### PR TITLE
Show a message if the viewer was blocked by the user

### PR DESCRIPTION
**Pull request by davidmz**
When the user A (blocked by the user B) browses the page of user B, show  the message: "Nothing to show here. Perhaps %screenName% has blocked you."

### CODE REVIEW

File: src/components/user-feed.jsx [hunk 1]
```diff
@@ -3,21 +3,31 @@ import React from 'react';
...
-        <p><b>{props.viewUser.screenName}</b> has a private feed.</p>
-      </div>
-    ) : (
-      <PaginatedView {...props}>
-        <Feed {...props} isInUserFeed={true}/>
-      </PaginatedView>
-    )}
-  </div>
-);
+export default props => {
```
Comment (indeyets) [inline src/components/user-feed.jsx]: Add a component name, while you're changing the code. `const UserFeed = (props) => …`

File: src/components/user-feed.jsx [hunk 2]
```diff
@@ -3,21 +3,31 @@ import React from 'react';
...
+  const {viewUser, location: {query}} = props;
+  const isBlocked = viewUser.blocked;
+  const isPrivate = viewUser.isPrivate === '1' && !viewUser.subscribed && !viewUser.isItMe;
+  const amIBlocked = viewUser.isPrivate === '0' && viewUser.statistics.posts !== '0' && props.visibleEntries.length === 0 && (!('offset' in query) || query.offset === '0');
```
Comment (indeyets) [inline src/components/user-feed.jsx]: shouldn't first part of condition be `!isPrivate` instead?

File: src/components/user-feed.jsx [hunk 3]
```diff
@@ -3,21 +3,31 @@ import React from 'react';
...
-      </div>
-    ) : (
-      <PaginatedView {...props}>
-        <Feed {...props} isInUserFeed={true}/>
-      </PaginatedView>
-    )}
-  </div>
-);
+export default props => {
+  const {viewUser, location: {query}} = props;
```
Comment (indeyets) [inline src/components/user-feed.jsx]: you can move destructuring right into component-function declaration: `({ viewUser, location: { query } }) => …`

File: src/components/user-feed.jsx [hunk 4]
```diff
@@ -3,21 +3,31 @@ import React from 'react';
...
+  const isBlocked = viewUser.blocked;
+  const isPrivate = viewUser.isPrivate === '1' && !viewUser.subscribed && !viewUser.isItMe;
+  const amIBlocked = viewUser.isPrivate === '0' && viewUser.statistics.posts !== '0' && props.visibleEntries.length === 0 && (!('offset' in query) || query.offset === '0');
+  return (
+    <div>
+      {isBlocked ? (
+        <div className="box-body">
+          <p>You have blocked <b>{viewUser.screenName}</b>, so all of their posts and comments are invisible to you.</p>
+          <p><a onClick={()=>props.userActions.unban({username: viewUser.username, id: viewUser.id})}>Un-block</a></p>
```
Comment (indeyets) [inline src/components/user-feed.jsx]: inline declaration of handlers is a bad pattern. No need to fix this here, but we should figure out something in the future

see https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-no-bind.md

**Review by indeyets — CHANGES_REQUESTED** [review]

File: src/components/user-feed.jsx (hunk 2 above)
Comment (davidmz) [reply src/components/user-feed.jsx]: isPrivate is a string, '0' or '1'. So I think it is better to explicitly compare it with string.

**Review by indeyets — APPROVED** [review]

**Pull request merged**